api-metastore/src/main/java/org/zalando/nakadi/service/SubscriptionService.java

```diff
@@ -153,6 +164,30 @@ public Subscription createSubscription(final SubscriptionBase subscriptionBase)
         return subscription;
     }
 
+    private Subscription createSubscriptionWithEventTypeLock(final SubscriptionBase subscriptionBase) {
+        try {
+            return transactionTemplate.execute(action -> {
+                eventTypeRepository.lockTable(EventTypeRepository.TableLock.ROW_EXCLUSIVE);
```

Comment:
On a second thought, a `SHARE` lock should suffice here, as long as the other transaction is taking the `ROW EXCLUSIVE` one.  This way concurrent requests to create subscriptions do not block each other (but block deletion of an event type).